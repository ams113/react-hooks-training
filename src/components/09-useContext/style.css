@import url('https://fonts.googleapis.com/css2?family=MuseoModerno:wght@300;400;500;600;700&display=swap');

/*? RESET */

* {
    margin: 0;
    padding: 0!important;
}

body {
    font-family: 'MuseoModerno', cursive;
}

/*? Main Navbar */

.btn {
   padding: .375rem !important;
}

.main-navbar{
    position: fixed;
    bottom: 0;
    bottom: 0;
    width: 100vw;
    height: 80px; /** ojo  */
    background-color: #23232e;
    text-decoration: none;
}

.navbar-container {
    list-style: none;
    display: flex;
    flex-direction: row;
    align-items: center !important;
    justify-content: space-around;
    height:  100%;
}

.logo {
    display: none;
}

/*? Items */

.navbar-item {
    width: 100%;
}

.navbar-link {
    display: flex;
    flex-direction: column;
    align-items: center; 
    justify-content: center;
    height: 80px; /** ojo  */
    text-decoration: none;
    filter: grayscale(100%) opacity(0.7);
    transition: 0.5s all ease;
}
/* color #551a8b */

.navbar-link:hover {
    filter: grayscale(0%) opacity(1)
}

.fa-primary {
    color: orange;
    height: 30px;
    transition: 0.6s all ease;
    
}

#dragon {
    color: #612f91;    
}
#fire {
    color: rgb(250, 41, 4);
}
#water {
    color: #1a84c4;
}
#wind {
    color: antiquewhite
}
#earth {
    color: mediumseagreen;
}

.text-link {
    display: none;
}

/*? MAIN */

.main-content  {
    padding: 10px !important;
}

@media only screen and (min-width: 600px) {

    main {
        margin-left: 100px !important;
    }

    .main-navbar {
        top:0;
        width: 100px;
        height: 100vh;
        transition: 0.5s all ease;


    }
    .main-navbar:hover {
        width: 300px;
    }

    .main-navbar:hover .text-link {
        display: block;
    }

    .navbar-container {
        
        flex-direction: column;
        justify-content: initial;
    }

    .navbar-item:last-child {
        margin-top: auto;
    }

    .navbar-link {
        flex-direction: row;
        justify-content: initial;
        height: 100px;
    }
    .navbar-link svg {
        margin: 0px 25px;
    }

    .navbar-link:hover {
        background-color: #141418;
    }
    .text-link {
        margin-left: 10px;
        color: violet;
    }
    .fa-primary {
        height: 40px;
        width: 40px;
    }
    .logo {
        display: block;
        font-weight: bold;
        text-transform: uppercase;
        margin-bottom: 10px;
        text-align: center;
        letter-spacing: 1px;
        width: 100%;
    }

    .logo svg{
        color: cadetblue;
    }
}