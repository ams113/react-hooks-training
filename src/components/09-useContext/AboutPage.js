import React, { useContext } from 'react'
import './style.css';
import { UserContext } from './UseContext';

export const AboutPage = () => {

    const { user, setUser } = useContext(UserContext);

    const handleClick = () => {
        setUser({});
    };
    
    return (
        <main className="main-content">
            <h1>AboutPage</h1>

            <hr />

            <pre>
                { JSON.stringify(user, null, 3 ) }
            </pre>
            
            <button 
                onClick={ handleClick }
                className="btn btn-warning">
                    Login
            </button>
        </main>
    )
}
