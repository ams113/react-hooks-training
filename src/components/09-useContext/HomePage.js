import React, { useContext } from 'react'
import { UserContext } from './UseContext'
import './style.css';
export const HomePage = () => {

    const { user } = useContext(UserContext);
  
    return (
        <main className="main-content">
            <h1>HomePage</h1>

            <pre>
                { JSON.stringify(user, null, 3 ) }
            </pre>
        </main>
    )
}
