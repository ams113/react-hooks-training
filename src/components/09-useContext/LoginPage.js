import React, { useContext } from 'react'
import { UserContext } from './UseContext';
import './style.css';

export const LoginPage = () => {

    const { setUser } = useContext(UserContext);

    return (
        <main className="main-content">
            <h1>LoginPage</h1>
            <hr />
            <button 
                onClick={ ()=> setUser({ id: 'asb', name: 'toni'})}
                className="btn btn-dark">
                    Login
            </button>
        </main>
    )
}
