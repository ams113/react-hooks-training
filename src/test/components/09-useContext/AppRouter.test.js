import React from 'react';
import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { UserContext } from '../../../components/09-useContext/UseContext';
import { LoginPage } from '../../../components/09-useContext/LoginPage';
import { AppRouter } from '../../../components/09-useContext/AppRouter';

describe('Testing Components: useContext/AppRouter', () => {

    const user = { id: 'asb', name: 'toni'};

    const wrapper = mount(
        <UserContext.Provider value={ { user } }> 
            <AppRouter />
        </UserContext.Provider>
    );

    test('should renders correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    
 
    
   

});