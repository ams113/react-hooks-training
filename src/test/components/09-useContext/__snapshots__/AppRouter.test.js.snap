// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Components: useContext/AppRouter should renders correctly 1`] = `
<div>
  <nav
    className="main-navbar"
  >
    <ul
      className="navbar-container"
    >
      <li
        className="logo"
      >
        <a
          className="navbar-link"
          href="/"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="fa-primary"
            data-icon="hat-wizard"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M496 448H16c-8.84 0-16 7.16-16 16v32c0 8.84 7.16 16 16 16h480c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16zm-304-64l-64-32 64-32 32-64 32 64 64 32-64 32-16 32h208l-86.41-201.63a63.955 63.955 0 0 1-1.89-45.45L416 0 228.42 107.19a127.989 127.989 0 0 0-53.46 59.15L64 416h144l-16-32zm64-224l16-32 16 32 32 16-32 16-16 32-16-32-32-16 32-16z"
              fill="currentColor"
            />
          </svg>
          <span
            className="text-link"
          >
            Menu
          </span>
        </a>
      </li>
      <li
        className="navbar-item"
      >
        <a
          aria-current={null}
          className="navbar-link"
          href="/home"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="fa-primary"
            data-icon="dragon"
            data-prefix="fas"
            focusable="false"
            id="dragon"
            role="img"
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18.32 255.78L192 223.96l-91.28 68.69c-10.08 10.08-2.94 27.31 11.31 27.31h222.7c-9.44-26.4-14.73-54.47-14.73-83.38v-42.27l-119.73-87.6c-23.82-15.88-55.29-14.01-77.06 4.59L5.81 227.64c-12.38 10.33-3.45 30.42 12.51 28.14zm556.87 34.1l-100.66-50.31A47.992 47.992 0 0 1 448 196.65v-36.69h64l28.09 22.63c6 6 14.14 9.37 22.63 9.37h30.97a32 32 0 0 0 28.62-17.69l14.31-28.62a32.005 32.005 0 0 0-3.02-33.51l-74.53-99.38C553.02 4.7 543.54 0 533.47 0H296.02c-7.13 0-10.7 8.57-5.66 13.61L352 63.96 292.42 88.8c-5.9 2.95-5.9 11.36 0 14.31L352 127.96v108.62c0 72.08 36.03 139.39 96 179.38-195.59 6.81-344.56 41.01-434.1 60.91C5.78 478.67 0 485.88 0 494.2 0 504 7.95 512 17.76 512h499.08c63.29.01 119.61-47.56 122.99-110.76 2.52-47.28-22.73-90.4-64.64-111.36zM489.18 66.25l45.65 11.41c-2.75 10.91-12.47 18.89-24.13 18.26-12.96-.71-25.85-12.53-21.52-29.67z"
              fill="currentColor"
            />
          </svg>
          <span
            className="text-link"
          >
            Invoke
          </span>
        </a>
      </li>
      <li
        className="navbar-item"
      >
        <a
          aria-current={null}
          className="navbar-link"
          href="/about"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="fa-primary"
            data-icon="fire"
            data-prefix="fas"
            focusable="false"
            id="fire"
            role="img"
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M216 23.86c0-23.8-30.65-32.77-44.15-13.04C48 191.85 224 200 224 288c0 35.63-29.11 64.46-64.85 63.99-35.17-.45-63.15-29.77-63.15-64.94v-85.51c0-21.7-26.47-32.23-41.43-16.5C27.8 213.16 0 261.33 0 320c0 105.87 86.13 192 192 192s192-86.13 192-192c0-170.29-168-193-168-296.14z"
              fill="currentColor"
            />
          </svg>
          <span
            className="text-link"
          >
            Fire
          </span>
        </a>
      </li>
      <li
        className="navbar-item"
      >
        <a
          aria-current={null}
          className="navbar-link"
          href="/login"
          onClick={[Function]}
        >
          <svg
            aria-hidden="true"
            className="fa-primary"
            data-icon="water"
            data-prefix="fas"
            focusable="false"
            id="water"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M562.1 383.9c-21.5-2.4-42.1-10.5-57.9-22.9-14.1-11.1-34.2-11.3-48.2 0-37.9 30.4-107.2 30.4-145.7-1.5-13.5-11.2-33-9.1-46.7 1.8-38 30.1-106.9 30-145.2-1.7-13.5-11.2-33.3-8.9-47.1 2-15.5 12.2-36 20.1-57.7 22.4-7.9.8-13.6 7.8-13.6 15.7v32.2c0 9.1 7.6 16.8 16.7 16 28.8-2.5 56.1-11.4 79.4-25.9 56.5 34.6 137 34.1 192 0 56.5 34.6 137 34.1 192 0 23.3 14.2 50.9 23.3 79.1 25.8 9.1.8 16.7-6.9 16.7-16v-31.6c.1-8-5.7-15.4-13.8-16.3zm0-144c-21.5-2.4-42.1-10.5-57.9-22.9-14.1-11.1-34.2-11.3-48.2 0-37.9 30.4-107.2 30.4-145.7-1.5-13.5-11.2-33-9.1-46.7 1.8-38 30.1-106.9 30-145.2-1.7-13.5-11.2-33.3-8.9-47.1 2-15.5 12.2-36 20.1-57.7 22.4-7.9.8-13.6 7.8-13.6 15.7v32.2c0 9.1 7.6 16.8 16.7 16 28.8-2.5 56.1-11.4 79.4-25.9 56.5 34.6 137 34.1 192 0 56.5 34.6 137 34.1 192 0 23.3 14.2 50.9 23.3 79.1 25.8 9.1.8 16.7-6.9 16.7-16v-31.6c.1-8-5.7-15.4-13.8-16.3zm0-144C540.6 93.4 520 85.4 504.2 73 490.1 61.9 470 61.7 456 73c-37.9 30.4-107.2 30.4-145.7-1.5-13.5-11.2-33-9.1-46.7 1.8-38 30.1-106.9 30-145.2-1.7-13.5-11.2-33.3-8.9-47.1 2-15.5 12.2-36 20.1-57.7 22.4-7.9.8-13.6 7.8-13.6 15.7v32.2c0 9.1 7.6 16.8 16.7 16 28.8-2.5 56.1-11.4 79.4-25.9 56.5 34.6 137 34.1 192 0 56.5 34.6 137 34.1 192 0 23.3 14.2 50.9 23.3 79.1 25.8 9.1.8 16.7-6.9 16.7-16v-31.6c.1-8-5.7-15.4-13.8-16.3z"
              fill="currentColor"
            />
          </svg>
          <span
            className="text-link"
          >
            Water
          </span>
        </a>
      </li>
      <li
        className="navbar-item"
      >
        <a
          className="navbar-link"
          href="#"
        >
          <svg
            aria-hidden="true"
            className="fa-primary"
            data-icon="wind"
            data-prefix="fas"
            focusable="false"
            id="wind"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M156.7 256H16c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h142.2c15.9 0 30.8 10.9 33.4 26.6 3.3 20-12.1 37.4-31.6 37.4-14.1 0-26.1-9.2-30.4-21.9-2.1-6.3-8.6-10.1-15.2-10.1H81.6c-9.8 0-17.7 8.8-15.9 18.4 8.6 44.1 47.6 77.6 94.2 77.6 57.1 0 102.7-50.1 95.2-108.6C249 291 205.4 256 156.7 256zM16 224h336c59.7 0 106.8-54.8 93.8-116.7-7.6-36.2-36.9-65.5-73.1-73.1-55.4-11.6-105.1 24.9-114.9 75.5-1.9 9.6 6.1 18.3 15.8 18.3h32.8c6.7 0 13.1-3.8 15.2-10.1C325.9 105.2 337.9 96 352 96c19.4 0 34.9 17.4 31.6 37.4-2.6 15.7-17.4 26.6-33.4 26.6H16c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16zm384 32H243.7c19.3 16.6 33.2 38.8 39.8 64H400c26.5 0 48 21.5 48 48s-21.5 48-48 48c-17.9 0-33.3-9.9-41.6-24.4-2.9-5-8.7-7.6-14.5-7.6h-33.8c-10.9 0-19 10.8-15.3 21.1 17.8 50.6 70.5 84.8 129.4 72.3 41.2-8.7 75.1-41.6 84.7-82.7C526 321.5 470.5 256 400 256z"
              fill="currentColor"
            />
          </svg>
          <span
            className="text-link"
          >
            Wind
          </span>
        </a>
      </li>
      <li
        className="navbar-item"
      >
        <a
          className="navbar-link"
          href="#"
        >
          <svg
            aria-hidden="true"
            className="fa-primary"
            data-icon="spa"
            data-prefix="fas"
            focusable="false"
            id="earth"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M568.25 192c-29.04.13-135.01 6.16-213.84 83-33.12 29.63-53.36 63.3-66.41 94.86-13.05-31.56-33.29-65.23-66.41-94.86-78.83-76.84-184.8-82.87-213.84-83-4.41-.02-7.79 3.4-7.75 7.82.23 27.92 7.14 126.14 88.77 199.3C172.79 480.94 256 480 288 480s115.19.95 199.23-80.88c81.64-73.17 88.54-171.38 88.77-199.3.04-4.42-3.34-7.84-7.75-7.82zM287.98 302.6c12.82-18.85 27.6-35.78 44.09-50.52 19.09-18.61 39.58-33.3 60.26-45.18-16.44-70.5-51.72-133.05-96.73-172.22-4.11-3.58-11.02-3.58-15.14 0-44.99 39.14-80.27 101.63-96.74 172.07 20.37 11.7 40.5 26.14 59.22 44.39a282.768 282.768 0 0 1 45.04 51.46z"
              fill="currentColor"
            />
          </svg>
          <span
            className="text-link"
          >
            Earth
          </span>
        </a>
      </li>
      <li
        className="navbar-item"
      >
        <a
          className="navbar-link"
          href="#"
        >
          <svg
            aria-hidden="true"
            className="fa-primary"
            data-icon="skull-crossbones"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M439.15 453.06L297.17 384l141.99-69.06c7.9-3.95 11.11-13.56 7.15-21.46L432 264.85c-3.95-7.9-13.56-11.11-21.47-7.16L224 348.41 37.47 257.69c-7.9-3.95-17.51-.75-21.47 7.16L1.69 293.48c-3.95 7.9-.75 17.51 7.15 21.46L150.83 384 8.85 453.06c-7.9 3.95-11.11 13.56-7.15 21.47l14.31 28.63c3.95 7.9 13.56 11.11 21.47 7.15L224 419.59l186.53 90.72c7.9 3.95 17.51.75 21.47-7.15l14.31-28.63c3.95-7.91.74-17.52-7.16-21.47zM150 237.28l-5.48 25.87c-2.67 12.62 5.42 24.85 16.45 24.85h126.08c11.03 0 19.12-12.23 16.45-24.85l-5.5-25.87c41.78-22.41 70-62.75 70-109.28C368 57.31 303.53 0 224 0S80 57.31 80 128c0 46.53 28.22 86.87 70 109.28zM280 112c17.65 0 32 14.35 32 32s-14.35 32-32 32-32-14.35-32-32 14.35-32 32-32zm-112 0c17.65 0 32 14.35 32 32s-14.35 32-32 32-32-14.35-32-32 14.35-32 32-32z"
              fill="currentColor"
            />
          </svg>
          <span
            className="text-link"
          >
            Exit
          </span>
        </a>
      </li>
    </ul>
  </nav>
  <main
    className="main-content"
  >
    <h1>
      HomePage
    </h1>
    <pre>
      {
   "id": "asb",
   "name": "toni"
}
    </pre>
  </main>
</div>
`;
